olmo/config.py

```diff
@@ -428,6 +434,29 @@ class ModelConfig(BaseConfig):
     See :data:`TrainConfig.precision` instead.
     """
 
+    def __post_init__(self):
+        if self.n_kv_heads is None:
+            self.n_kv_heads = self.n_heads
+
+    @classmethod
+    def update_legacy_settings(cls, config: D) -> D:
+        new_config = config.copy()
+        if om.is_dict(new_config):
+            assert isinstance(new_config, DictConfig)
+
+            if hasattr(new_config, "multi_query_attention"):
```

Comment:
I am probably missing sth but since ModelConfig is a dataclass won't it always have the attribute multi_query_attention? So maybe this should rather check if multi_query_attention is set to True and and n_kv_heads is set and error out in that case?